### PR TITLE
Created templates for engage case study and webinar heros

### LINKED ISSUE #4404: /engage Landing page template - build

### PR DESCRIPTION
**Pull request by pmahnke**
## Done

- Created templates for heros for engage pages for case studies and webinars

## QA

- Check out this feature branch
- Run the site using the command `./run serve`
- You will need to setup a test page or add one of the below includes on a page in the /engage directory to test these

#### no image or class

```{% include "engage/shared/_header-case-study.html" with title="Allan Gray unlocks unprecedented availability and productivity with Ubuntu K8s"   %}```

html
```
<section class="p-strip--light">
  <div class="row u-equal-height">
    <div class="col-8">
      <h1>
        Allan Gray unlocks unprecedented availability and productivity with Ubuntu K8s
      </h1>
    </div>
  </div>
</section>
```

image

#### image, no class

```{% include "engage/shared/_header-case-study.html" with title="Allan Gray unlocks unprecedented availability and productivity with Ubuntu K8s"   %}```

html

```
<section class="p-strip--light">
  <div class="row u-equal-height">
    <div class="col-7">
      <h1>
        Allan Gray unlocks unprecedented availability and productivity with Ubuntu K8s
      </h1>
    </div>
    <div class="col-5 u-align--center u-vertically-center">
      <img src="https://assets.ubuntu.com/v1/071489b4-K8s+logo+mobile.svg" alt="image for (( h1 ))">
    </div>
  </div>
</section>
```
image

#### image, class

```{% include "engage/shared/_header-case-study.html" with title="Allan Gray unlocks unprecedented availability and productivity with Ubuntu K8s"  image="071489b4-K8s+logo+mobile.svg" class="test-class" %}```

html
```
<section class="p-strip test-class">
  <div class="row u-equal-height">
    <div class="col-7">
      <h1>
        Allan Gray unlocks unprecedented availability and productivity with Ubuntu K8s
      </h1>
    </div>
    <div class="col-5 u-align--center u-vertically-center">
      <img src="https://assets.ubuntu.com/v1/071489b4-K8s+logo+mobile.svg" alt="image for (( h1 ))">
    </div>
  </div>
</section>
```

image

#### title, subtitle, date, cta, class

```
{% include "engage/shared/_header-webinar.html" with title="What's New in Ubuntu 18.10"  image="d9105909-Dell_XPS_Laptop_Left-Desktop.png?w=413" subtitle="Canonical product managers discuss the new Ubuntu 18.10 release features in this on-demand webinar" date="Webinar is live on 3 December 2018 at 8pm GMT" url="#register-section" cta="Watch the webinar" class="test-class" %}
```

```
<section class="p-strip test-class">
  <div class="row u-equal-height">
    <div class="col-7">
      <h1 style="font-weight: 100;">
        What's New in Ubuntu 18.10
      </h1>
      <h4>
        Canonical product managers discuss the new Ubuntu 18.10 release features in this on-demand webinar
      </h4>
      <p>
        Webinar is live on 3 December 2018 at 8pm GMT
      </p>
      <p>
        <a href="#register-section" class="p-button--neutral">
          Watch the webinar
        </a>
      </p>
    </div>
    <div class="col-5 u-align--center u-vertically-center">
      <img src="https://assets.ubuntu.com/v1/d9105909-Dell_XPS_Laptop_Left-Desktop.png?w=413" alt="image for (( title ))">
    </div>
  </div>
</section>
```
image

#### no date

```
{% include "engage/shared/_header-webinar.html" with title="What's New in Ubuntu 18.10"  image="d9105909-Dell_XPS_Laptop_Left-Desktop.png?w=413" subtitle="Canonical product managers discuss the new Ubuntu 18.10 release features in this on-demand webinar" url="#register-section" cta="Watch the webinar" %}
```

```
<section class="p-strip--light">
  <div class="row u-equal-height">
    <div class="col-7">
      <h1 style="font-weight: 100;">
        What's New in Ubuntu 18.10
      </h1>
      <h4>
        Canonical product managers discuss the new Ubuntu 18.10 release features in this on-demand webinar
      </h4>
      
      <p>
        <a href="#register-section" class="p-button--neutral">
          Watch the webinar
        </a>
      </p>
    </div>
    <div class="col-5 u-align--center u-vertically-center">
      <img src="https://assets.ubuntu.com/v1/d9105909-Dell_XPS_Laptop_Left-Desktop.png?w=413" alt="image for (( title ))">
    </div>
  </div>
</section>
```
image

#### no image, no class

```
{% include "engage/shared/_header-webinar.html" with title="What's New in Ubuntu 18.10" url="#register-section" cta="Watch the webinar"  %}
```

```
{% include "engage/shared/_header-webinar.html" with title="What's New in Ubuntu 18.10"  subtitle="Canonical product managers discuss the new Ubuntu 18.10 release features in this on-demand webinar" date="Webinar is live on 3 December 2018 at 8pm GMT" url="#register-section" cta="Watch the webinar" %}
```

```
<section class="p-strip--light">
  <div class="row u-equal-height">
    <div class="col-8">
      <h1 style="font-weight: 100;">
        What's New in Ubuntu 18.10
      </h1>
      <h4>
        Canonical product managers discuss the new Ubuntu 18.10 release features in this on-demand webinar
      </h4>
      <p>
        Webinar is live on 3 December 2018 at 8pm GMT
      </p>
      <p>
        <a href="#register-section" class="p-button--neutral">
          Watch the webinar
        </a>
      </p>
    </div>
    
  </div>
</section>
```
image

#### no subtitle, no image, no class

```
<section class="p-strip--light">
  <div class="row u-equal-height">
    <div class="col-8">
      <h1 style="font-weight: 100;">
        What's New in Ubuntu 18.10
      </h1>
      
      
      <p>
        <a href="#register-section" class="p-button--neutral">
          Watch the webinar
        </a>
      </p>
    </div>
    
  </div>
</section>
```
image

## Issue / Card

Fixes #4404

### CODE REVIEW

File: templates/engage/shared/_header-case-study.html [hunk 1]
```diff
@@ -0,0 +1,12 @@
+<section class="{% if class %}p-strip {{ class }}{% else %}p-strip--light{% endif %}">
+  <div class="row u-equal-height">
+    <div class="{% if image %}col-7{% else %}col-8{% endif %}">
+      <h1 style="font-weight: 100;">
+        {{ title }}
+      </h1>
+    </div>
+    {% if image %}<div class="col-5 u-align--center u-vertically-center">
+      <img src="{% if image != "http" %}{{ ASSET_SERVER_URL }}{% endif %}{{ image }}" alt="image for (( title ))" >
```
Comment (deadlight) [inline templates/engage/shared/_header-case-study.html]: Nice

**Comment by anasereijo** [comment]
Thanks @pmahnke for doing this.
Just some small changes and I think it's ready to go 👍 

#### No image or class

- [ ] because there is just a title and no copy, we can remove the `margin-bottom` from the `h1` by using `class="u-no-margin--bottom"`


#### Image, no class

- [ ] vertically align the title by using `class="row u-equal-height u-vertically-center"`

- [ ] once again, because there is just a title and no copy, we can remove the `margin-bottom` from the `h1` by using `class="u-no-margin--bottom"`

#### Image and class

- [ ] vertically align the title by using `class="row u-equal-height u-vertically-center"`

- [ ] once again, because there is just a title and no copy, we can remove the `margin-bottom` from the `h1` by using `class="u-no-margin--bottom"`

#### Webinar style 
- [ ] vertically align the title by using `class="row u-equal-height u-vertically-center"`

#### Webinar style, no date
- [ ] vertically align the title by using `class="row u-equal-height u-vertically-center"`

#### Webinar style - no image, no class
- [ ] vertically align the title by using `class="row u-equal-height u-vertically-center"`

**Comment by pmahnke** [comment]
@anasereijo I think I have done this... seems like

1. class="row u-equal-height u-vertically-center" on all title rows
2. class="u-no-margin--bottom" on all case study h1's

File: templates/engage/shared/_header-webinar.html [hunk 2]
```diff
@@ -0,0 +1,23 @@
+<section class="{% if class %}p-strip {{ class }}{% else %}p-strip--light{% endif %}">
+  <div class="row u-equal-height u-vertically-center">
```
Comment (anthonydillon) [inline templates/engage/shared/_header-webinar.html]: u-vertically-center should be on the col

File: templates/engage/shared/_header-webinar.html [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+    <div class="{% if image %}col-7{% else %}col-8{% endif %}">
+      <h1 style="font-weight: 100;">
```
Comment (anthonydillon) [inline templates/engage/shared/_header-webinar.html]: Maybe a bespoke styling to remove the margin bottom of this is the only-child. Save a lot of null checking logic.

**Comment by caldav** [comment]
Small request about the CTA, could we have it on both templates and could we make it so that it only appears on mobile? Because it's mostly used as a fast track to skip the copy and go straight to the form/video. Eg. https://www.ubuntu.com/engage/developer-desktop

**Comment by pmahnke** [comment]
case studies

image

webinars

image

**Comment by anasereijo** [comment]
@caldav I really like your idea of always having a CTA on mobile! 👍

**Comment by deadlight** [comment]
@pmahnke could you add a file with all of the examples demoed and then rebuild the demo? It's hard for design to look at all of the options.

Something like [this file](https://gist.github.com/deadlight/793f669cd24f5e93ed6275dc48838db0) should work

**Comment by pmahnke** [comment]
OK, I have addressed all comments I think... and added a test page [/engage/_test_header](https://www-ubuntu-com-canonical-websites-pr-4450.run.demo.haus/engage/_test_header)

One thing to note, :only-child will not know that a cta is hidden in large/med, so there will be margin-bottom in this case (see the last example on the test page)

**Comment by caldav** [comment]
I think it looks great. Just two things:

- There is a lack of spacing between subtitle and CTA when there is no date (which would be typically the case for promoting a webinar after it aired).
- If the CTA is the only difference between both templates, and since it can be declared optionally, maybe we don't need two templates? (cf. https://github.com/canonical-websites/www.ubuntu.com/pull/4450#issuecomment-442984325)

**Comment by pmahnke** [comment]
Updated to a single include and added a permanent help page [/engage/_heading-help][https://www-ubuntu-com-canonical-websites-pr-4450.run.demo.haus/engage/_heading-help]

**Pull request merged**